open_rarity/resolver/testset_resolver.py

```diff
@@ -116,63 +115,24 @@ def get_token_ids(
         token_ids = get_token_ids(batch_id)
         logger.debug(
             f"Starting batch {batch_id} for collection "
-            f"{collection_with_metadata.opensea_slug}: "
-            f"Processing {len(token_ids)} tokens"
+            f"{slug}: Processing {len(token_ids)} tokens"
         )
 
-        try:
-            assets = fetch_opensea_assets_data(
-                slug=collection_with_metadata.opensea_slug, token_ids=token_ids
-            )
-        except Exception as e:
-            logger.exception(
-                "FAILED: get_assets: could not fetch opensea assets for %s: %s",
-                token_ids,
-                e,
-                exc_info=True,
-            )
-            break
+        tokens = get_tokens_from_opensea(
+            opensea_slug=collection_with_metadata.opensea_slug,
+            token_ids=token_ids,
+        )
 
         # We will store all rarities calculated across providers in this list
-        tokens_rarity_batch: list[TokenWithRarityData] = []
-        for asset in assets:
-            token_metadata = opensea_traits_to_token_metadata(
-                asset_traits=asset["traits"]
-            )
-            asset_contract_address = asset["asset_contract"]["address"]
-            asset_contract_type = asset["asset_contract"][
-                "asset_contract_type"
-            ]
-            if asset_contract_type == "non-fungible":
-                token_standard = TokenStandard.ERC721
-            elif asset_contract_type == "semi-fungible":
-                token_standard = TokenStandard.ERC1155
-            else:
-                raise Exception(
-                    f"Unexpected asset contrat type: {asset_contract_type}"
-                )
-
-            token_with_rarity = TokenWithRarityData(
-                token=Token(
-                    token_identifier=EVMContractTokenIdentifier(
-                        identifier_type="evm_contract",
-                        contract_address=asset_contract_address,
-                        token_id=asset["token_id"],
-                    ),
-                    token_standard=token_standard,
-                    metadata=token_metadata,
-                ),
-                rarities=[],
-            )
-
-            tokens_rarity_batch.append(token_with_rarity)
+        tokens_rarity_batch = list(
+            map(lambda t: TokenWithRarityData(token=t, rarities=[]), tokens)
+        )
```

Comment:
A list comp is a little more pythonic and consistent with the rest of the library unless we plan to use the generator that `map` produces.
```python
tokens_rarity_batch = [TokenWithRarityData(token=t, rarities=[]) for t in tokens]
```